contracts/finality/src/exec/finality.rs

```diff
@@ -347,16 +338,15 @@ fn slash_finality_provider(
         )
         .map_err(|err| ContractError::SecretKeyExtractionError(err.to_string()))?;
 
-    // Emit slashing event.
-    // Raises slashing event to babylon over IBC.
-    let msg = ExecuteMsg::Slashing {
-        sender: info.sender.clone(),
-        evidence: evidence.clone(),
-    };
-    let wasm_msg: WasmMsg = WasmMsg::Execute {
-        contract_addr: env.contract.address.to_string(),
-        msg: to_json_binary(&msg)?,
-        funds: vec![],
+    let msg = BabylonMsg::EquivocationEvidence {
+        signer: info.sender.to_string(),
+        fp_btc_pk: evidence.fp_btc_pk.clone(),
+        block_height: evidence.block_height,
+        pub_rand: evidence.pub_rand.clone(),
+        canonical_app_hash: evidence.canonical_app_hash.clone(),
+        fork_app_hash: evidence.fork_app_hash.clone(),
+        canonical_finality_sig: evidence.canonical_finality_sig.clone(),
+        fork_finality_sig: evidence.fork_finality_sig.clone(),
     };
```

Comment:
Not sure if I'm using the clone fields here properly.

Reply:
The clone here is correct. Those are needed because otherwise we are taking ownership of these variables under `evidence` and compiler will be unhappy